src/credentials/FabricTable.cpp

```diff
@@ -445,21 +445,35 @@ CHIP_ERROR FabricTable::VerifyCredentials(const ByteSpan & noc, const ByteSpan &
 }
 
 const FabricInfo * FabricTable::FindFabric(const Crypto::P256PublicKey & rootPubKey, FabricId fabricId) const
+{
+    return FindFabricCommon(rootPubKey, fabricId);
+}
+
+const FabricInfo * FabricTable::FindIdentity(const Crypto::P256PublicKey & rootPubKey, FabricId fabricId, NodeId nodeId) const
+{
+    return FindFabricCommon(rootPubKey, fabricId, nodeId);
+}
+
+const FabricInfo * FabricTable::FindFabricCommon(const Crypto::P256PublicKey & rootPubKey, FabricId fabricId, NodeId nodeId) const
 {
     P256PublicKey candidatePubKey;
 
     // Try to match pending fabric first if available
     if (HasPendingFabricUpdate())
     {
         bool pubKeyAvailable = (mPendingFabric.FetchRootPubkey(candidatePubKey) == CHIP_NO_ERROR);
-        if (pubKeyAvailable && rootPubKey.Matches(candidatePubKey) && fabricId == mPendingFabric.GetFabricId())
+        auto tmpNodeId       = (nodeId == kUndefinedNodeId) ? mPendingFabric.GetNodeId() : nodeId;
```

Comment:
Suggest making this a NodeId explicitly instead  of auto. Also `tmpNodeId` is a bad name: what is it for? Why is it "temp" ?

Is there a way to make this common logic less tricky or more self-explanatory?

Reply:
We can name it `matchingNodeId`. But I don't have better suggestions on making this logic less tricky. I'm open to suggestions.